libs/permission-checker/tests/Check/RunnerSyncApi/CanResolveSharedCodeAndVariablesTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\PersmissionChecker\Tests\Check\RunnerSyncApi;
+
+use Generator;
+use Keboola\PermissionChecker\Check\RunnerSyncApi\CanResolveSharedCodeAndVariables;
+use Keboola\PermissionChecker\Exception\PermissionDeniedException;
+use Keboola\PermissionChecker\StorageApiToken;
+use PHPUnit\Framework\TestCase;
+
+class CanResolveSharedCodeAndVariablesTest extends TestCase
+{
+    public function testValidPermissionsCheck(): void
+    {
+        $expectedComponentIds = [
+            'keboola.shared-code',
+            'keboola.variables',
+        ];
+
+        $tokenMock = $this->createMock(StorageApiToken::class);
+        $tokenMock->expects(self::exactly(2))
+            ->method('hasAllowedComponent')
+            ->willReturnCallback(function (string $componentId) use (&$expectedComponentIds) {
+                self::assertSame(
+                    array_shift($expectedComponentIds),
+                    $componentId
+                );
+                return true;
+            });
```

Comment:
Ten mock mi prijde zbytecny. To, ze to bez jedne z tech komponent neprojde mas otestovany ve `testInvalidPermissionsCheck`